src/Carousel/Carousel.tsx

```diff
@@ -1,64 +1,140 @@
-        touchMove: true,
-        slidesToShow: 1,
-        slidesToScroll: 1,
-        autoplay: props.autoPlay ? true : false,
-        autoplaySpeed: props.autoPlaySpeed ? props.autoPlaySpeed : 3000,
-        afterChange: props.afterChange,
-    };
+export const Carousel: React.FC<CarouselProps> = React.memo(({ afterChange, transitionDuration, transitionStyle, infinite, showIndicators, list = [], ...props }: CarouselProps) => {
+    const [active, setActive] = React.useState<number>(0);
+    const [size, setSize] = React.useState<number>(0);
+    const [nav, setNav] = React.useState<NavigationDirection>("next");
+    const [id, setId] = React.useState<string>("");
+    const [blocked, setBlocked] = React.useState<boolean>(false);
+    const [className, setClassName] = React.useState<string>("carousel");
+
+    /** ----- Utilities ----- */
+
+    /**
+     * Handles navigating to a slide
+     * @param {NavigateTrigger} e Navigation trigger event
+     */
+    const goToSlide: (e: NavigateTrigger) => void = React.useCallback(
+        (e: NavigateTrigger) => {
+            e.preventDefault();
+            let newActive: number;
+            let newNav: NavigationDirection;
+            const isInfinite: boolean = infinite || infinite === undefined;
+            if (!blocked) {
+                if (e.currentTarget.tagName === "LI") {
+                    /** Indicator clicked */
+                    const slideTo: number = Number(e.currentTarget.dataset.slideTo);
+                    newNav = slideTo > active ? "next" : "prev";
+                    newActive = slideTo;
+                } else {
+                    /** Navigation controls clicked or keyboard event happened */
+                    if (e.type === "click") {
+                        newNav = e.currentTarget.dataset.slide as NavigationDirection;
+                    } else {
+                        switch ((e as React.KeyboardEvent<HTMLAnchorElement>).key.toLowerCase()) {
+                            case "arrowleft":
+                                newNav = "prev";
+                                break;
+                            case "arrowright":
+                                newNav = "next";
+                                break;
+                            case "space":
+                            case " ":
+                                newNav = e.currentTarget.dataset.slide as NavigationDirection;
+                        }
+                    }
+                    switch (newNav) {
+                        case "prev":
+                            newActive = active === 0 ? (isInfinite ? size - 1 : newActive) : active - 1;
+                            break;
+                        case "next":
+                            newActive = active === size - 1 ? (isInfinite ? 0 : newActive) : active + 1;
+                            break;
+                    }
+                }
+                if (newNav !== undefined && newActive !== undefined) {
```

Comment:
not really important but maybe good in case you have more variables that needs to check for undefined.
it is supported since ie9
```suggestion
                if ([newNav, newActive].every((el: any) => el !== undefined) {
```